controllers/suite_test.go

```diff
@@ -60,98 +61,88 @@ func init() {
 }
 
 func TestMain(m *testing.M) {
-	fmt.Println("Creating a new test environment")
-	env = envtest.New()
-
-	// Set up the MachineNodeIndex
-	if err := noderefutil.AddMachineNodeIndex(ctx, env.Manager); err != nil {
-		panic(fmt.Sprintf("unable to setup machine node index: %v", err))
-	}
+	setupIndexes := func(ctx context.Context, mgr ctrl.Manager) {
+		// Set up the MachineNodeIndex
+		if err := noderefutil.AddMachineNodeIndex(ctx, mgr); err != nil {
+			panic(fmt.Sprintf("unable to setup machine node index: %v", err))
+		}
 
-	// Set up the MachineProviderIDIndex
-	if err := noderefutil.AddMachineProviderIDIndex(ctx, env.Manager); err != nil {
-		panic(fmt.Sprintf("unable to setup machine providerID index: %v", err))
+		// Set up the MachineProviderIDIndex
+		if err := noderefutil.AddMachineProviderIDIndex(ctx, mgr); err != nil {
+			panic(fmt.Sprintf("unable to setup machine providerID index: %v", err))
+		}
 	}
 
-	// Set up a ClusterCacheTracker and ClusterCacheReconciler to provide to controllers
-	// requiring a connection to a remote cluster
-	tracker, err := remote.NewClusterCacheTracker(
-		env.Manager,
-		remote.ClusterCacheTrackerOptions{
-			Log: log.Log,
-			Indexes: []remote.Index{
-				{
-					Object:       &corev1.Node{},
-					Field:        noderefutil.NodeProviderIDIndex,
-					ExtractValue: noderefutil.IndexNodeByProviderID,
+	setupReconcilers := func(ctx context.Context, mgr ctrl.Manager) {
+		// Set up a ClusterCacheTracker and ClusterCacheReconciler to provide to controllers
+		// requiring a connection to a remote cluster
+		tracker, err := remote.NewClusterCacheTracker(
+			mgr,
+			remote.ClusterCacheTrackerOptions{
+				Log: log.Log,
```

Comment:
Should we use mgr.GetLogger() here?

Reply:
In our production code we use `ctrl.Log.WithName("remote").WithName("ClusterCacheTracker")`
Would it make sense to use the same in our tests?